lib/model/queue_test.go

```diff
@@ -12,56 +12,21 @@ import (
 	"time"
 
 	"github.com/d4l3k/messagediff"
+	"github.com/syncthing/syncthing/lib/config"
 )
 
 func TestJobQueue(t *testing.T) {
 	// Some random actions
-	q := newJobQueue()
-	q.Push("f1", 0, time.Time{})
-	q.Push("f2", 0, time.Time{})
-	q.Push("f3", 0, time.Time{})
-	q.Push("f4", 0, time.Time{})
+	q := newJobQueue(config.OrderAlphabetic, "")
+	for i := 4; i > 0; i-- {
+		q.Push(fmt.Sprintf("f%d", i), 0, time.Time{})
+	}
 
 	progress, queued := q.Jobs()
 	if len(progress) != 0 || len(queued) != 4 {
 		t.Fatal("Wrong length")
 	}
 
-	for i := 1; i < 5; i++ {
-		n, ok := q.Pop()
-		if !ok || n != fmt.Sprintf("f%d", i) {
-			t.Fatal("Wrong element")
-		}
-		progress, queued = q.Jobs()
-		if len(progress) != 1 || len(queued) != 3 {
-			t.Log(progress)
-			t.Log(queued)
-			t.Fatal("Wrong length")
-		}
-
-		q.Done(n)
-		progress, queued = q.Jobs()
-		if len(progress) != 0 || len(queued) != 3 {
-			t.Fatal("Wrong length", len(progress), len(queued))
-		}
-
-		q.Push(n, 0, time.Time{})
-		progress, queued = q.Jobs()
-		if len(progress) != 0 || len(queued) != 4 {
-			t.Fatal("Wrong length")
-		}
-
-		q.Done("f5") // Does not exist
-		progress, queued = q.Jobs()
-		if len(progress) != 0 || len(queued) != 4 {
-			t.Fatal("Wrong length")
-		}
-	}
-
-	if len(q.progress) > 0 || len(q.queued) != 4 {
-		t.Fatal("Wrong length")
-	}
-
```

Comment:
Why does this disappear?

Reply:
Because it repeatedly adds and removes items, which we never do in code and which isn't supported by `diskoverflow.Sorted` (once you start getting/removing items, you can't add new ones).